src/Datasource/EntityTrait.php

```diff
@@ -119,6 +119,16 @@ trait EntityTrait
      */
     protected $_registryAlias = '';
 
+    /**
+     * Set to true in your entity's class definition or
+     * via application logic. When true. has() and related
+     * methods will use `array_key_exists` instead of `isset`
+     * to decide if fields are 'defined' in an entity.
+     *
+     * @var bool
+     */
+    protected $_hasAllowsNull = false;
```

Comment:
Do we need to make this easier to enable globally or do we expect users to enable one at a time?

Reply:
Looks ok to me. I couldn't think of a better name. It requires some understanding of the change either way.

I guess we should deprecate it immediately though for 5.0, right?

Reply:
> I guess we should deprecate it immediately though for 5.0, right?

Effectively yes. The property would be deprecated in 5.x with the goal of changing the default behavior for 6.0.